R/remotes.R

```diff
@@ -394,6 +473,110 @@ renv_remotes_resolve_github <- function(entry) {
 
 }
 
+renv_remotes_resolve_git <- function(parsed) {
+
+  # handle refs that are direct refs
+  # handle potential PR refs
+  # handle tags too?
+
+
+  # For now, set package & version to dummy values.
+  # will need to clone first to resolve these.
+  temp_package <- parsed$repo
+  temp_version <- "UNKNOWN"
+  repo         <- parsed$repo
+  url          <- parsed$url
+  subdir       <- parsed$subdir
+
+  # handle git ref
+  pull <- parsed$pull %||% ""
+  ref <- parsed$ref %||% ""
+
+  remote_ref <- case(
+    nzchar(pull) ~ renv_remotes_resolve_git_pull(pull),
+    nzchar(ref)  ~ ref,
+    TRUE         ~ ""
+  )
+
+  record <- list(
+    Package        = temp_package,
+    Version        = temp_version,
+    Source         = "git",
+    RemoteType     = "git",
+    RemoteUrl      = url,
+    RemoteSubdir   = subdir,
+    RemoteRef      = remote_ref
+  )
+
+  desc <- renv_remotes_resolve_git_description(record)
+
+  record$Package <- desc$Package
+  record$Version <- desc$Version
+
+  record
+}
+
+
+renv_remotes_resolve_git_sha_ref <- function(record) {
+  renv_git_preflight()
+  template <- c(
+    "git ls-remote \"${ORIGIN}\" \"${REF}\" | head -1 | sed 's/\t.*//'"
```

Comment:
This may not work on Windows (where `head` and `sed` are not necessarily available / on the PATH).

Could we just use the result of `git ls-remote \"${ORIGIN}\" \"${REF}\"` and then filter or transform the output in R?

Reply:
I had the same thought while adapting it! (& ignored it for some reason) 

I agree it makes more sense to rely on R to manipulate the output string